### DISCUSSION TITLE
How to apply force to rigid bodies?

### DISCUSSION THREAD

**Discussion by drifter089** [pmndrs/react-three-rapier#176] [Q&A] [answered]
I can;t figure out how to apply force to rigid bodies

**Comment by isaac-mason** [comment, marked as answer]
You can apply forces to a rigid body using the ref of a RigidBody component.

e.g.

```ts
const ExampleRigidBodyComponent = (props) => {
  const ref = useRef()

  useEffect(() => {
    ref.current.addForce({ x: 0, y: 1, z: 0 }, true)
  }, [])

  return (
    <RigidBody ref={ref}>
      { /* ... */ }
    </RigidBody>
  )
}
```

Here's a related sandbox that shows how to use a rigid body component's ref (in this example an impulse is applied on clicking a box)

https://codesandbox.io/s/react-three-rapier-applyimpulse-8nuh3m

**Reply by drifter089** [reply]
thank you very much I was passing it as an array

**Reply by Seanmclem** [reply]
Where is the documentation about this? I didn't find much.
